GOVERNANCE.md

```diff
@@ -175,39 +195,108 @@ The project Steward is currently: **@FredKSchott**
+
+TSC members are guardians over the Astro codebase. Their duty is to ensure code quality, correctness and security. 
 
-Responsibilities will vary. Most often, a staff member will regularly meet the responsibilites of either the [Maintainer (L2)](#level-2---maintainer) or [Core Maintainer (L3)](#level-3---core-mainainer) role.
+A TSC member guides the direction of the project and ensures a healthy future for the Astro codebase. TSC members are ultimately responsible for technical decision making when it comes to any changes to the Astro codebase.
+
+A TSC member has significant sway in software design decisions. For this reason, coding experience is critical for this role. TSC membership is one of the only roles that requires a significant contribution history of code to the Astro project on GitHub.
+
+#### Privileges
+
+- `@tsc` role on [Discord](https://astro.build/chat)
+- Invitation to the private `#tsc` channel on Discord
+- Invitation to the `tsc` team on GitHub.
+- Ability to merge all GitHub PRs.
+- Ability to vote on RFCs and technical initiatives (see [Voting](#voting) below).
+
+#### Responsibilities 
+
+- Participating in RFC discussions and technical meetings.
+- Assisting with design and implementation of non-trivial GitHub PRs.
+- Reviewing and merging larger, non-trivial PRs.
+- Maintaining and improving overall codebase architecture.
+- Tracking and ensuring progress of open pull requests.
+- Mentoring and guiding other community contributors.
 
 #### Nomination
 
-There is no nomination process for this role. The project steward is responsible for granting and revoking the `@staff` role.
+- To be nominated, a nominee is expected to already be active in technical discussions and performing some of the responsibilities of a TSC member.
+- You can be nominated by any existing Core member (L3 or above). Note: This includes all existing TSC members as well.
+- Once nominated, there will be a vote by existing Core members.
+- See [vote rules & requirements](#voting) for info on how this vote works.
+
+
+### Staff
+
+**Staff** is a special designation for employees of [The Astro Technology Company](https://astro.build/company). 
+
+#### Privileges
 
-#### Leaving Staff
+- `@staff` role on [Discord](https://astro.build/chat)
+- New name color on Discord: **yellow**.
+- Invitation to some private channels on Discord, at the discretion of the project Steward.
+- Invitation to the `staff` team on GitHub.
 
-When someone leaves the Astro Technology Company, they lose staff privileges and return to their original membership level in our governance structure (whatever level they were at before joining staff).
+Staff membership does not grant any additional abilities when it comes to voting and project governance. A Staff member is still eligible for other roles in the community and may still vote as defined by their other roles. For example, a Staff member who is also a part of `@core` will be able to vote as any other `@core` member would.
```

Comment:
Not really! This flow you're describing is mainly covered by the "Core In Residence" role. 
For 99% of people, I imagine it will work like this:

- Someone joins The Astro Technology Company to work on the core codebase full-time
- They are nominated for "Core In Residence", meaning that they get the `@core` status but can not participate in voting.
- They leave the company and lose their core status. If they want to stick around our community, they can go through the normal nomination channels.

If someone joins the company but does not work in the community, they may get staff but wouldn't even be nominationed for maintainer, core, etc.

If someone joins the company and contributes to the codebase for many years, becoming a leader in the community, then they could have their "in residence" limitations removed by a normal core nomination vote.

Hope that helps clarify!

Reply:
Ah, so basically this is kinda switching the Staff role to a relatively decorative role + "core in residence" is an option to fast-track permissions for someone in the company but who hasn't necessarily been around long enough to earn any of the other statuses?

Reply:
yup, exactly.